### PR TITLE
MM-15638 Adding authentication for monitoring apps and general code improvememts

### PR DESCRIPTION
**Pull request by stylianosrigas**
Jira Ticket -> https://mattermost.atlassian.net/browse/MM-15638

- Github authentication using oauth2 added for:
  - Kibana
  - Grafana
  - Prometheus
- Code cleanup

### CODE REVIEW

**Comment by crspeller** [comment]
@stylianosrigas Not sure I am currently qualified to review this. If you have some time to walk me though the repository and the changes that would be beneficial.

**Comment by stylianosrigas** [comment]
@crspeller Yeap of course. Lets have a talk :)

File: chart-values/elasticsearch_values.yaml [hunk 1]
```diff
@@ -111,7 +111,7 @@ client:
     minAvailable: 1
     # maxUnavailable: 1
   ingress:
-    enabled: true
+    enabled: false
```
Comment (cpanato) [inline chart-values/elasticsearch_values.yaml]: you can set to false and delete the other values, just keep the enabled: false

File: chart-values/kibana_values.yaml [hunk 2]
```diff
@@ -66,15 +66,16 @@ service:
...
   annotations:
     kubernetes.io/ingress.class: nginx
     certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+    nginx.ingress.kubernetes.io/auth-url: "https://auth.example.com/oauth2/auth"
+    nginx.ingress.kubernetes.io/auth-signin: "https://auth.example.com/oauth2/start?rd=https://$host$request_uri"
   tls:
     - secretName: kibana-server-tls
       hosts:
-        - kibana-central.mattermost.com
+        - kibana.example.com
```
Comment (cpanato) [inline chart-values/kibana_values.yaml]: this is correct?

File: chart-values/elasticsearch_values.yaml (hunk 1 above)
Comment (stylianosrigas) [reply chart-values/elasticsearch_values.yaml]: @cpanato removed the non needed section.

File: chart-values/kibana_values.yaml (hunk 2 above)
Comment (stylianosrigas) [reply chart-values/kibana_values.yaml]: @cpanato Yeap, I thought its better to keep some example values in these chart values as the central monitoring ones are being stored in another repo.

**Pull request merged**